tlair/src/main/scala/at/forsyte/apalache/tla/lir/transformations/impl/TrackerWithListeners.scala

```diff
@@ -8,15 +9,43 @@ import at.forsyte.apalache.tla.lir.transformations._
   *
   * For any input x, track(t)(x) and t(x) are equal.
   */
-sealed case class TrackerWithListeners( listeners : TransformationListener* )
-  extends TransformationTracker {
-  override def track( transformation : TlaExTransformation ) : TlaExTransformation = {
-    ex =>
-      val newEx = transformation( ex )
-      listeners foreach {
-        _.onTransformation( ex, newEx )
-      }
-      newEx
+sealed case class TrackerWithListeners(listeners: TransformationListener*)
+    extends TransformationTracker {
+  override def track(
+      transformation: TlaExTransformation
+  ): TlaExTransformation = { ex =>
+    val newEx = transformation(ex)
+    listeners foreach {
+      _.onTransformation(ex, newEx)
+    }
+    newEx
   }
 
+  /**
+    * Decorate a declaration transformation with a tracker.
+    *
+    * @param transformation a declaration transformation
+    * @return a new declaration transformation that applies tr and tracks the changes
+    */
+  override def trackDecl(
+      transformation: TlaDeclTransformation
+  ): TlaDeclTransformation = { decl =>
+    val newDecl = transformation(decl)
+    listeners foreach { _.onDeclTransformation(decl, newDecl) }
+    newDecl
+  }
+
+  /**
+    * A specialized version of trackDecl, which is tuned to TlaOperDecl.
+    *
+    * @param transformation a declaration transformation
+    * @return a new declaration transformation that applies tr and tracks the changes
+    */
+  override def trackOperDecl(
```

Comment:
Same as above, please explain the need for `TlaOperDecl`-specific methods.

Reply:
Added an explanation to TransformationTracker